proto/hummock.proto

```diff
@@ -109,6 +109,20 @@ message UncommittedEpoch {
   repeated SstableInfo tables = 2;
 }
 
+message VnodeWatermark {
+  bytes watermark = 1;
```

Comment:
The size of `watermark` is unbounded, right?

Reply:
Yes. It's a key without vnode. Basically the same size of key range in sstable info.

Reply:
> the current range delete will have the same problem when this happens

Yes. The difference is it results in SST files in object store and busy compaction tasks, while this PR results in huge HummockVersion in meta store.